qds_sdk/clusterv2.py

```diff
@@ -0,0 +1,550 @@
+from qds_sdk.qubole import Qubole
+from qds_sdk.resource import Resource
+from qds_sdk.cloud.cloud import Cloud
+from qds_sdk.engine import Engine
+from qds_sdk import util
+import argparse
+
+import sys
+import json
+
+def str2bool(v):
+    return v.lower() in ("yes", "true", "t", "1")
+
+class ClusterCmdLine:
+
+    @staticmethod
+    def parsers(args):
+        action = args[0]
+        argparser = argparse.ArgumentParser(
+            prog="qds.py cluster",
+            description="Cluster Operations for Qubole Data Service.")
+        subparsers = argparser.add_subparsers(title="Cluster operations")
+
+        create = subparsers.add_parser("create", help="Create a new cluster")
+        if action == "create":
+            ClusterCmdLine.create_update_clone_parser(create, action="create")
+            create.set_defaults(func=ClusterV2.create)
+
+        update = subparsers.add_parser("update", help="Update the settings of an existing cluster")
+        if action == "update":
+            ClusterCmdLine.create_update_clone_parser(update, action="update")
+            update.set_defaults(func=ClusterV2.update)
+
+        clone = subparsers.add_parser("clone", help="Clone a cluster from an existing one")
+        if action == "clone":
+            ClusterCmdLine.create_update_clone_parser(clone, action="clone")
+            clone.set_defaults(func=ClusterV2.clone)
+
+        return argparser
+
+
+    @staticmethod
+    def create_update_clone_parser(subparser, action=None):
+
+        # cloud config parser
+        cloud = Cloud.get_cloud_object()
+        cloud.cloud_config_parser(subparser)
+
+        # cluster info parser
+        ClusterInfoV2.cluster_info_parser(subparser, action)
+
+        # engine config parser
+        Engine().engine_parser(subparser)
+
+
+    @staticmethod
+    def run(args):
+        parser = ClusterCmdLine.parsers(args)
+        arguments = parser.parse_args(args)
+        customer_ssh_key = util._read_file(arguments.customer_ssh_key_file, "customer ssh key file")
+
+        # This will set cluster info and monitoring settings
```

Comment:
Do we need to write parser for cluster actions like list, start, terminate, status, add_node etc ?